plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java

```diff
@@ -99,7 +99,6 @@
  * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
  */
 @Extension
-@Restricted(NoExternalUse.class)
```

Comment:
I think all the methods that you aren't interested in using should be marked as noexternaluse they were previously covered by the class level restriction.

i.e. all the web methods (doXXX) should be restricted

Reply:
A few more that I think should be restricted?
https://github.com/jenkinsci/configuration-as-code-plugin/blob/fc25779bd0d76de937c4ce4d75e3e076664f1047/plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java#L330
https://github.com/jenkinsci/configuration-as-code-plugin/blob/fc25779bd0d76de937c4ce4d75e3e076664f1047/plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java#L273
https://github.com/jenkinsci/configuration-as-code-plugin/blob/fc25779bd0d76de937c4ce4d75e3e076664f1047/plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java#L632
https://github.com/jenkinsci/configuration-as-code-plugin/blob/fc25779bd0d76de937c4ce4d75e3e076664f1047/plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java#L794
https://github.com/jenkinsci/configuration-as-code-plugin/blob/fc25779bd0d76de937c4ce4d75e3e076664f1047/plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java#L808

Also not so sure if public APIs should be exposed without documentation but I guess it can be improved, i.e. https://github.com/jenkinsci/configuration-as-code-plugin/blob/fc25779bd0d76de937c4ce4d75e3e076664f1047/plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java#L451

and not is an outputstream as an input parameter what you want? or would you want a better API like a string as a return value?

Reply:
It is fine with the OutputStream